src/index.rs

```diff
@@ -111,14 +170,16 @@ impl ResourceIndex {
                                 );
                                 false
                             }
-                            Ok(curr_modified) => curr_modified > prev_modified,
+                            Ok(curr_modified) => {
+                                curr_modified > SystemTime::from(prev_modified)
+                            }
                         },
                     }
                 }
             })
             .collect();
 
-        let mut deleted: HashSet<ResourceId> = HashSet::new();
+        let mut deleted = HashMap::new();
 
         // treating deleted and updated paths as deletions
         prev_paths
```

Comment:
_Maybe `indexed_path` would be a better name._

Reply:
_Observation:_
preserved_paths = prev_paths \/ curr_paths
prev_paths - preserved_paths = prev_path - curr_path _= deleted?_

_Renaming suggestions:_
prev_paths -> index_paths
curr_paths -> fs_paths
preserved_paths -> common_paths